sdks/python/apache_beam/transforms/core.py

```diff
@@ -864,10 +872,15 @@ def from_runner_api_parameter(pardo_payload, context):
     fn, args, kwargs, si_tags_and_types, windowing = pickler.loads(
         pardo_payload.do_fn.spec.payload)
     if si_tags_and_types:
-      raise NotImplementedError('deferred side inputs')
+      raise NotImplementedError('explicit side input data')
     elif windowing:
       raise NotImplementedError('explicit windowing')
-    return ParDo(fn, *args, **kwargs)
+    result = ParDo(fn, *args, **kwargs)
+    indexed_side_inputs = [
+        (int(ix[4:]), pvalue.AsSideInput.from_runner_api(si, context))
+        for ix, si in pardo_payload.side_inputs.items()]
+    result.side_inputs = [si for _, si in sorted(indexed_side_inputs)]
```

Comment:
<!--new_thread; commit:3dfe862220afbc5e57d7ac674a329cb5e66164cf; resolved:0-->
do we need to sort?

Reply:
Yes, added a comment.